### DISCUSSION TITLE
nuxt layers with private repositories

### DISCUSSION THREAD

**Discussion by ilyam2014** [nuxt/nuxt#26066] [Questions] [answered]
can be layer used with private github repository?

**Comment by AloisSeckar** [comment, marked as answer]
What I did (for GitLab, but it should be roughly the same):

1) published layer to our company private repo as `@my/layer`

2) created personal access token on my GitLab account

3) in target project set following values in `.env`:
```
GIGET_REGISTRY=https://<<PRIVATE-REPO>>/api/v4/projects/<<PROJECT-ID>>/packages/npm/
GIGET_AUTH=<<TOKEN>>
```

4) added the dependency into target project `package.json`:
```
"devDependencies": {
    "@my/layer": "latest"
}
```

5) in `nuxt.config.ts`:
```
extends: [
    ['@my/layer', { install: true }]
]
```

This works for me. 

[In the docs](https://nuxt.com/docs/guide/going-further/layers) they mention possibility to add `gitlab:` (or `github:`) prefix to `extends`, but that doesn't work for me for some reason.

**Reply by ilyam2014** [reply]
thank you

**Comment by ErosBurastero** [comment]
Hi there, i would like to ask you about this, i've installed my private package and extended it successfully, but when i try to use the components it doesn't find them, i've set all my components outside .playground directory

i got errors like this: Failed to resolve component: TextField
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 

Have you ever seen something like this? thank you

**Reply by ErosBurastero** [reply]
I also get this error: [Vue warn]: Component <Anonymous> is missing template or render function.

**Reply by AloisSeckar** [reply]
Not sure if this is your case, but I had found out, that if you changed default behavior of Nuxt auto-import of components in any way in base layer, you need to repeat the same config in the extending nuxt.config.ts as well.
